rule/contract.go

```diff
@@ -0,0 +1,67 @@
+package rule
+
+// Type represents the type of a typed expression.  Any expression has
+// a return type, and some expressions also receive typed parameters.
+type Type int
+
+// These constants represent the complete set of abstract types usable
+// in expressions within Regula.  Not that these abstract types don't
+// necessary equate directly to concrete types that you might think
+// about within Go.  In particular "NUMBER" and "ANY" exist to define
+// parameters that do not have to be of a fixed type.
+const (
+	BOOLEAN Type = iota
+	STRING
+	INTEGER
+	FLOAT
+	NUMBER // A special type that can be promoted to INTEGER or FLOAT
+	ANY    // A special type that can be promoted to any other.
+)
+
+// Cardinality expresses the number of times a term might be repeated
+// in an expression.
+type Cardinality int
+
+// These constants define the complete set of cardinality possible for
+// terms of a contract.  We can expect a term to occur exactly once
+// (ONE), or zero, one or many times (MANY).
+const (
+	MANY = 0
+	ONE  = 1
+)
+
+type Term struct {
```

Comment:
:tulip: Would be cool to have a comment for this one too, the rest is so well commented

Reply:
... oops missed that one.  Resolved